test/mocks/mock_server.ex

```diff
@@ -160,14 +160,43 @@ defmodule ExActivecampaign.MockServer do
   defp list_get_success(conn) do
     conn
     |> Plug.Conn.send_resp(
-         200,
-         Poison.encode!(%{
-           list: %{
-             stringid: "example-list",
-             name: "Example List",
-             id: "1"
-           }
-         })
-       )
+      200,
+      Poison.encode!(%{
+        list: %{
+          stringid: "example-list",
+          name: "Example List",
+          id: "1"
+        }
+      })
+    )
+  end
+
+  post "/v1" do
+    case conn.query_params do
+      %{"api_action" => "contact_sync", "api_output" => "json"} -> v1_post_contact_sync_json(conn)
+    end
+  end
+
+  def v1_post_contact_sync_json(conn) do
+    case conn.body_params do
+      %{
+        "email" => "johndoe@example.com",
+        "first_name" => "John",
+        "last_name" => "Doe",
+        "phone" => "7223224241"
+      } ->
+        conn
+        |> Plug.Conn.send_resp(
+          200,
+          Poison.encode!(%{
+            subscriber_id: 1,
+            sendlast_should: 0,
+            sendlast_did: 0,
+            result_code: 1,
+            result_message: "Contact added",
+            result_output: "json"
+          })
+        )
+    end
```

Comment:
Might it be worth a test for the update side of contact sync as well as adding? And perhaps then retrieving the contact with V3 to make sure they play nicely together... possibly overkill though

Reply:
Ooo, good spot - I'll get some in there :)

Reply:
Hmm, thinking about it, I'm not sure I'll be able to achieve this without adding some kind of in-memory store to hold onto the initial create, and subsequently handle the update - at which point, I guess we'd be testing the implementation of the in-memory store more than the actual library logic.

I guess as the tests here are more Unit-y than functional, these kind of tests could fall more into the integrating codebase's remit (until, that is, I've found a more solid solution!)

Reply:
Ah yes good point. Is there a sandbox account set up? https://developers.activecampaign.com/page/developer-sandbox-accounts

I wonder whether that would be suitable for automated tests (deleting contacts each time so as not to exceed the limits) - if so it would probably make more sense to use that for tests within Kraken rather than in this repo.

Reply:
Wow, no - I am completely baffled as to how I didn't even realise that was a thing! Have sent a request for one now!! 👀